### PR TITLE
Improve members inputs

### PR DESCRIPTION
**Pull request by corentingiraud**
### Description of your PR

- [x] Merge member-new and member-edit component
- [x] Auto correct user input (only on front side, we can discuss if we schould verify / correct on back side)
- [x] Create AVAILABLE SCHOOL constant and select input. Same comment as above for back side verification.
- [x] Add autofocus
- [x] Remove code for CREATE UPDATE & REGISTER. The code is required for DELETE.
- [x] Distinguish re-membership from new membership. Implement a member search.
- [ ] Update script for current value

### Pre-review TODO

- [x] PR is rebased on top of `origin/master` 
- [ ] PR provide new tests for new behaviors
- [x] PR doesn't introduce commented out code
- [x] PR has been tested locally on the main use cases
- [ ] Documentation have been updated if necessary

### CODE REVIEW

File: client/src/app/members/new-edit/member-new-edit.component.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  nameFormatter(name: string): string {
+    const newName = !name ? name :
+      name.replace(/[\wÀ-ÿ]+(\S&-)*/g, txt => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase())
+      .replace(/[`~!@#$%^&*()_|+\=?;:'",.<>\{\}\[\]\\\/]/gi, '');
+    return newName;
+  }
+
+  search(): void {
+    this.searchQuery = this.nameFormatter(this.searchQuery);
+    this.notEnoughCharacters = this.searchQuery.length <= 3 ? true : false;
```
Comment (Botweiser) [inline client/src/app/members/new-edit/member-new-edit.component.ts]: Code Smell Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=307&issues=AWZCLTrRvOM6zLh9TXFQ&open=AWZCLTrRvOM6zLh9TXFQ)

[ISSUE_KEY]: AWZCLTrRvOM6zLh9TXFQ

File: server/utils/helpers.js [hunk 2]
```diff
@@ -68,6 +68,17 @@ const RANGE_SCHEMA = Joi.object({
...
 
+/**
+ * This constant is used by Joi to validate query from a request.
+ * It represent an object with a search query.
+ * This query must contain at least 4 characters
+ *
+ * @type {ObjectSchema} Joi schema
+ */
+const SEARCH_SCHEMA = Joi.object({
+  q: Joi.string().regex(/[a-z -]{4,}/i),
```
Comment (Embraser01) [inline server/utils/helpers.js]: It require at least 4 chars? 3 may be better (for name like *Léa*) ?

File: server/app/services/member-service.js [hunk 3]
```diff
@@ -205,6 +207,30 @@ async function deleteMember(memberId) {
...
+    Sequelize.fn('concat', Sequelize.col('firstName'), ' ', Sequelize.col('lastName')),
+    'like',
+    `%${query}%`,
+  );
+  const members = await Member.findAll({
+    where,
+    include: [{
+      model: Registration,
+      as: 'registrations',
+      attributes: ['year', 'createdAt'],
```
Comment (Embraser01) [inline server/app/services/member-service.js]: An order by maybe? Also I discovered `scopes` in sequelize and it looks really nice! http://docs.sequelizejs.com/manual/tutorial/scopes.html

File: client/src/app/members/new-edit/member-new-edit.component.ts (hunk 1 above)
Comment (Embraser01) [inline client/src/app/members/new-edit/member-new-edit.component.ts]: Same here, can we set the min chars to 3 ?

File: server/app/services/member-service.js [hunk 4]
```diff
@@ -205,6 +207,30 @@ async function deleteMember(memberId) {
...
+ * @param query {string} query.
+ * @return {Promise<Member[]>} Members which match the query
+ */
+async function searchMembers(query) {
+  logger.verbose('Member service: searching members with query %s', query);
+  const where = Sequelize.where(
+    Sequelize.fn('concat', Sequelize.col('firstName'), ' ', Sequelize.col('lastName')),
+    'like',
+    `%${query}%`,
+  );
```
Comment (Embraser01) [inline server/app/services/member-service.js]: To do search queries, we should probably use https://www.w3resource.com/mysql/mysql-full-text-search-functions.php in order to optimize queries

File: client/src/app/members/new-edit/member-new-edit.component.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
```
Comment (Botweiser) [inline client/src/app/members/new-edit/member-new-edit.component.ts]: Code Smell Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=307&issues=AWZCLTrRvOM6zLh9TXFQ&open=AWZCLTrRvOM6zLh9TXFQ)

[ISSUE_KEY]: AWZCLTrRvOM6zLh9TXFQ

File: server/app/services/member-service.js (hunk 4 above)
Comment (Embraser01) [reply server/app/services/member-service.js]: @kistora We will do it later, it working for now (with +3000 members)

**Pull request merged**